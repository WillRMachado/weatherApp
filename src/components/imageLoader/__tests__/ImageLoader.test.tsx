import React from 'react';
import Image from 'react-native-image-progress';
import screenRender from '../../../../jestHelpers/screenRender';
import screenSnapshotCreator from '../../../../jestHelpers/screenSnapshotCreator';
import ImageLoader from '../ImageUriLoader';

describe('ImageLoader Component', () => {
  const mockUri = 'mockedUri';
  const ImageLoaderWithProps = () => <ImageLoader imageUri={mockUri} />;

  it('renders correctly', async () => {
    const component = await screenRender(ImageLoaderWithProps);
    expect(component).toBeTruthy();
  });

  it('matches snapshot', async () => {
    const component = await screenSnapshotCreator(ImageLoaderWithProps);
    expect(component).toMatchSnapshot();
  });

  it('gets path properly', async () => {
    const component = await screenRender(ImageLoaderWithProps);
    const imageLoader = component.root.findAllByType(Image)[0];
    const imageLoaderSource = imageLoader.props.source;
    expect(imageLoaderSource.uri).toEqual(mockUri);
  });
});
