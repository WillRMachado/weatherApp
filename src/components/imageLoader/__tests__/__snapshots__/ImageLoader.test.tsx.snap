// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageLoader Component matches snapshot 1`] = `null`;
